Makefile.common

```diff
@@ -21,9 +21,10 @@ ifeq ($(SYSTEM), win32)
   endif
   CSMLCSCODE = $(shell cygpath -w $(CSMLDIR)/csml.cs)
   EXECPREFIX=
-
+  PREFIX_DLL=
 else
-  CSC=gmcs
+  CSC=mcs
   CSMLCSCODE = $(CSMLDIR)/csml.cs
   EXECPREFIX=LD_LIBRARY_PATH=.
+  PREFIX_DLL=lib
```

Comment:
As far as I understand, the reason for this is as follows. On Windows, a library named "m" will generally be in a file "m.dll". On Linux, it would be in a file "libm.so" (or a variation thereof). So, when you write `[DllImport("m")]`, .NET on Windows will transform that into "m.dll", and Mono on Linux will transform that into "libm.so". In case of CSML, an explicit extension is applied, and Mono does not add a second ".so", but it still adds "lib".

The precise set of rules for mangling library names is long and very messy; it is described in detail on this [Mono documentation page](http://www.mono-project.com/docs/advanced/pinvoke/). I'll quote the gist of it:

> How do you write portable Platform Invoke (P/Invoke) code that will work cross-platform?
> The short answer is that you don’t.

It appears that, indeed, `[DllImport("foo")]` would do the right thing on both Windows (append ".dll" as long as "foo" doesn't contain any dots) and Linux (append ".so" and don't prepend "lib"), so I could try doing that.

I don't have a Windows OCaml development environment on hand (I cross-compile everything with my [opam-cross-windows](https://github.com/whitequark/opam-cross-windows) library), so could you test such a change for me once I do it?